backend/layers/processing/schema_migration.py

```diff
@@ -109,8 +109,7 @@ def collection_migrate(self, collection_id: str, collection_version_id: str, can
         if not any(dataset.metadata.schema_version != current_schema_version for dataset in version.datasets):
             # check if there are datasets to migrate.
             return {
-                "can_publish": str(False),  # skip publishing, because the collection is already published and no
-                # revision is created, or the collection is private or a revision.
+                "can_publish": str(can_publish),
```

Comment:
I don't think this will work to address the publish concern. can_publish will be set to False in gather_collections if the collection is in a revision or private. So, this will always be False for any publishable collection created outside of this run of this function